File: code-signal-arcade-universe/python3/adjacent_elements_product.py
# Original solution

def adjacent_elements_product(l):
    """
    List->Integer, given a list, return the largest product of consecutive pairs
    """
    res = float('-inf')
    for i in range(0, len(l) - 1):
        if (l[i] * l[i + 1] > res):
            res = l[i] * l[i + 1]
    return res

# More elegant solution

def max_product_pair(l):
    """
    List->Integer, given a list, return the largest product of consecutive pairs
    """
    return max([l[i] * l[i+1] for i in range(0, len(l)-1)])

File: code-signal-arcade-universe/intro/clojure/adjacent_elements_product.clj
;; Original Solution
(defn adjacent-elements-product
  "Vector->Integer, returns the largest product of consecutive pairs in a vector"
  [v]
  (apply max (for [i (range (- (count v) 1))]
               (* (v i) (v (+ i 1))))))

;; Alternative solution

(defn adjacent-elements-product
  "Vector->Integer, returns the largest product of consecutive pairs in a vector"
  [v]
  (apply max (map #(apply * %) (partition 2 1 v))))
